core/sr-io/without_std.rs

```diff
@@ -331,8 +331,116 @@ pub mod ext {
+
+		/// Sign a piece of data with authority key.
+		///
+		/// `sig_data` has to be a pointer to a slice of 64 bytes.
+		/// Returns `0` if successful, nonzero otherwise.
+		fn ext_sign(data: *const u8, len: u32, sig_data: *mut u8) -> u32;
+
+		/// Returns current UNIX timestamp (milliseconds)
+		fn ext_timestamp() -> u64;
+
+		/// Pause execution until given timestamp (milliseconds; `deadline`) is reached.
+		///
+		/// The deadline is obtained by querying the current timestamp via `ext_timestamp`
+		/// and then adding some time to it.
+		fn ext_sleep_until(deadline: u64);
+
+		/// Generate a random seed
+		///
+		/// `data` has to be a pointer to a slice of 32 bytes.
+		fn ext_random_seed(data: *mut u8);
+
+		/// Write a value to local storage.
+		fn ext_local_storage_set(key: *const u8, key_len: u32, value: *const u8, value_len: u32);
+
+		/// Read a value from local storage.
+		///
+		/// if `value_len` is u32::max_value the value has not been found.
+		fn ext_local_storage_get(key: *const u8, key_len: u32, value_len: *mut u32) -> *mut u8;
+
+		/// Initiaties a http request.
+		///
+		/// Returns a `RequestId(u16)` of initiated request, any value beyond `u16::max_value`
+		/// signifies an error.
+		///
+		/// `meta` is parity-codec encoded additional parameters to the request (like redirection policy,
+		/// timeouts, certificates policy, etc). The format is not yet specified and the field is currently
+		/// only reserved for future use.
+		fn ext_http_request_start(
+			method: *const u8,
+			method_len: u32,
+			url: *const u8,
+			url_len: u32,
+			meta: *const u8,
+			meta_len: u32
+		) -> u32;
+
+		/// Add a header to the request.
+		///
+		/// Returns `0` if successful, nonzero otherwise.
+		fn ext_http_request_add_header(
+			request_id: u32,
+			name: *const u8,
+			name_len: u32,
+			value: *const u8,
+			value_len: u32
+		) -> u32;
+
+		/// Write a chunk of request body.
+		///
+		/// Writing an empty chunks finalises the request.
+		/// Passing `0` as deadline blocks forever.
+		/// Returns `0` if successful, nonzero otherwise.
+		fn ext_http_request_write_body(
+			request_id: u32,
+			chunk: *const u8,
+			chunk_len: u32,
+			deadline: u64
+		) -> u32;
+
+		/// Block and wait for the responses for given requests.
+		///
+		/// Note that if deadline is not provided the method will block indefinitely,
+		/// otherwise unready responses will produce `DeadlineReached` status.
+		/// (see #primitives::offchain::HttpRequestStatus)
+		///
+		/// Make sure that `statuses` have the same length as ids.
+		/// Passing `0` as deadline blocks forever.
+		fn ext_http_response_wait(
+			ids: *const u32,
+			ids_len: u32,
+			statuses: *mut u32,
+			deadline: u64
+		);
+
+		/// Read all response headers.
+		///
+		/// Note the headers are only available before response body is fully consumed.
+		/// Returns parity-codec encoded vector of pairs `(HeaderKey, HeaderValue)`.
+		/// In case invalid `id` is passed it returns an empty vector.
```

Comment:
Tbh, I'd avoid the vector terminology here, because one could mistakingly assume that it is a `Vec` or can be safely treated as so (it can't: see my other response about UB invoked here).

Reply:
Yes, thanks!